wire/relay.go

```diff
@@ -64,16 +66,24 @@ func (p *Relay) Close() error {
 }
 
 // Cache enables caching of messages that don't match any consumer. They are
-// only cached if they match the given predicate, within the given context.
-func (p *Relay) Cache(ctx context.Context, predicate Predicate) {
+// only cached if they match the given predicate.
+func (p *Relay) Cache(predicate *Predicate) {
 	p.mutex.Lock()
 	defer p.mutex.Unlock()
 
 	if p.IsClosed() {
 		return
 	}
 
-	p.cache.Cache(ctx, predicate)
+	p.cache.Cache(predicate)
+}
+
+// ReleaseCache disable caching for the given predicate.
+func (p *Relay) ReleaseCache(predicate *Predicate) {
```

Comment:
Should we add the release method to the `wire.Cache` interface ? Because, I think the behavior of the Cache interface has changed.

Earlier, the caching of version zero updates was skipped when the context passed to `Accept` has been cancelled.
Now, contexts are not used for disabling cache. Instead, an explicit `Release` call is used.

It looks like a better approach to me. Also, could you document this change in the commit message ?

Reply:
`Cache` is not an interface. You mean `Cacher`? However, `Cacher` by its name is intended to only deal with caching, not releasing. I think the terminology might be mixed up now. It might make sense to refactor this a bit. I am not even sure if there is any use in having the interface, as we only use one implementation and this part is also not exposed to the outside as far as I can see.

If you have a concrete suggestion on what to change (maybe remove the interface?) feel free to create an issue for that. But I think we should rather do it in a separate PR, if you do such a change at all.

Reply:
Yes I meant the `Cacher` interface. I thought we should include `Release` method in the `Cacher` interface because it complements the `Cache` method, by disabling caching of a particular item. Looking further, I noticed
1. `Cacher` interface is referenced in the client package. Here, we use only the `Cache` method and not the `Release` method. 
2. `Cache` is the only implementation of `Cacher` interface. The `Release` method on it is an implementation detail, that is referenced only inside the wire package.

 
So, my suggestion would be that, in a separate PR:
1. We can make `Cache`, its methods and the `MakeCache` function unexported.
2. We can retain the `Cacher` interface, because it allows us to freely refactor the `Cache` implementation (like removing use of context and adding Release method, which you did in this PR).

Reply:
Created #286